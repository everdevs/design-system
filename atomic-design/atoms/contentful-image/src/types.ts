import { Breakpoints, Palette, PropsWithTheme } from "@evernest/theme";
import React from "react";

export interface ImgWrapperProps extends PropsWithTheme {
	bgColor: keyof Palette;
}

export interface ContentfulImageProps extends React.HTMLProps<HTMLDivElement> {
	bgColor?: keyof Palette;
}

export type ImageSizes = {
	[key in keyof Breakpoints]: keyof Breakpoints;
};

export enum AssetFit {
	pad = "pad",
	fill = "fill",
	scale = "scale",
	crop = "crop",
	thumb = "thumb",
}

export enum AssetFocus {
	center = "center",
	top = "top",
	right = "right",
	left = "left",
	bottom = "bottom",
	"top_right" = "top_right",
	"top_left" = "top_left",
	"bottom_right" = "bottom_right",
	"bottom_left" = "bottom_left",
	face = "face",
	faces = "faces",
}

export enum AssetFormat {
	jpg = "jpg",
	png = "png",
	webp = "webp",
}

export interface AssetOptions {
	fm?: AssetFormat;
	fl?: "progressive";
	h?: number;
	w?: number;
	q?: number;
	f?: AssetFocus;
	fit?: AssetFit;
}

export interface BuildSrcSetParams {
	url: string;
	sizes: ImageSizes;
	fm: AssetFormat;
	breakpoints: Breakpoints;
}
