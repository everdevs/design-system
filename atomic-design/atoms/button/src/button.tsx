import { css } from "@emotion/core";
import styled from "@emotion/styled";
import { ButtonProps, Size, Variation } from "./types";

const Button = styled("button", {
	shouldForwardProp: (propName: string) => !["theme"].includes(propName),
})<ButtonProps>`
	cursor: pointer;
	border: 0;
	${({ theme, size, variation }) => css`
		box-shadow: ${theme.tokens.shadow.medium.css};
		height: ${size === Size.small ? "20px" : "40px"};
	`};
`;

Button.defaultProps = {
	type: "button",
	size: Size.medium,
	variation: Variation.default,
};

export { Button };
