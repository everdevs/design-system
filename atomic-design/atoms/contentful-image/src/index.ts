export * from "./contentful-image";
export * from "./utils";
export * from "./types";
