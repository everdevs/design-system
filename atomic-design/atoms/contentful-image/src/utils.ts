import { stringify as queryStringify } from "query-string";
import { AssetOptions } from "./types";

export const getImageUrl = (url: string, options: AssetOptions): string =>
	`${url}?${queryStringify(options)}`;
