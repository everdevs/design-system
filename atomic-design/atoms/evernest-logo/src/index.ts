export * from "./evernest-logo";
export * from "./types";
