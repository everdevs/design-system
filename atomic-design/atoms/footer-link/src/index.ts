export * from "./footer-link";
export * from "./types";
