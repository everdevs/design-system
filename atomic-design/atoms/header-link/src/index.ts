export * from "./header-link";
export * from "./types";
