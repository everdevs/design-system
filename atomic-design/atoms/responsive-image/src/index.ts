export * from "./responsive-image";
export * from "./utils";
export * from "./types";
