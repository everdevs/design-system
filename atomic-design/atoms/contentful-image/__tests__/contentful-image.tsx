import { createTree, Wrapper } from "@evernest/dev-helpers";
import { mount } from "enzyme";
import React from "react";
import { ContentfulImage } from "../src";

const imageArgs = {
	src:
		"//images.ctfassets.net/vij0xfkj6nfm/65B2DBSyhcTksD4H9nuGds/a3be81862610240ee35085e63835f531/007-web-1000px.jpg",
	width: 1000,
	height: 667,
	alt: "placeholder image",
};

test("ContentfulImage renders", () => {
	const tree = createTree(
		<Wrapper>
			<ContentfulImage {...imageArgs} />
		</Wrapper>
	);
	expect(tree).toMatchSnapshot();
});

test("ContentfulImage mounts", () => {
	const wrapper = mount(
		<Wrapper>
			<ContentfulImage {...imageArgs} />
		</Wrapper>
	);
	expect(wrapper.find(ContentfulImage));
});

test("ContentfulImage calls onLoad", () => {
	const callback = jest.fn(() => {});
	const wrapper = mount(
		<Wrapper>
			<ContentfulImage {...imageArgs} onLoad={callback} />
		</Wrapper>
	);
	wrapper.find("img").at(1).simulate("load");
	expect(callback.mock.calls.length).toBe(1);
});
