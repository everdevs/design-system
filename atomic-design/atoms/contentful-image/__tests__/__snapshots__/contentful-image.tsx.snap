// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentfulImage renders 1`] = `
<div
  className="css-1mo6rv1"
>
  <span
    className="css-6zpqpl"
  >
    <img
      alt="placeholder image"
      className="css-bk0hd7"
      height={667}
      loading="lazy"
      src="//images.ctfassets.net/vij0xfkj6nfm/65B2DBSyhcTksD4H9nuGds/a3be81862610240ee35085e63835f531/007-web-1000px.jpg?fl=progressive&fm=jpg&q=10&w=320"
      width={1000}
    />
    <picture
      className="css-snvx98"
    >
      <source
        sizes="(min-width: 1200px) 1200px,(min-width: 1200px) 1200px,(min-width: 768px) 768px,320px"
        srcSet="//images.ctfassets.net/vij0xfkj6nfm/65B2DBSyhcTksD4H9nuGds/a3be81862610240ee35085e63835f531/007-web-1000px.jpg?fm=webp&w=320 320w,//images.ctfassets.net/vij0xfkj6nfm/65B2DBSyhcTksD4H9nuGds/a3be81862610240ee35085e63835f531/007-web-1000px.jpg?fm=webp&w=768 768w,//images.ctfassets.net/vij0xfkj6nfm/65B2DBSyhcTksD4H9nuGds/a3be81862610240ee35085e63835f531/007-web-1000px.jpg?fm=webp&w=1200 1200w,//images.ctfassets.net/vij0xfkj6nfm/65B2DBSyhcTksD4H9nuGds/a3be81862610240ee35085e63835f531/007-web-1000px.jpg?fm=webp&w=1200 1200w"
        type="image/webp"
      />
      <source
        sizes="(min-width: 1200px) 1200px,(min-width: 1200px) 1200px,(min-width: 768px) 768px,320px"
        srcSet="//images.ctfassets.net/vij0xfkj6nfm/65B2DBSyhcTksD4H9nuGds/a3be81862610240ee35085e63835f531/007-web-1000px.jpg?fl=progressive&fm=jpg&w=320 320w,//images.ctfassets.net/vij0xfkj6nfm/65B2DBSyhcTksD4H9nuGds/a3be81862610240ee35085e63835f531/007-web-1000px.jpg?fl=progressive&fm=jpg&w=768 768w,//images.ctfassets.net/vij0xfkj6nfm/65B2DBSyhcTksD4H9nuGds/a3be81862610240ee35085e63835f531/007-web-1000px.jpg?fl=progressive&fm=jpg&w=1200 1200w,//images.ctfassets.net/vij0xfkj6nfm/65B2DBSyhcTksD4H9nuGds/a3be81862610240ee35085e63835f531/007-web-1000px.jpg?fl=progressive&fm=jpg&w=1200 1200w"
        type="image/jpeg"
      />
      <img
        alt="placeholder image"
        className="css-6s2dw2"
        height={667}
        loading="lazy"
        onLoad={[Function]}
        src="//images.ctfassets.net/vij0xfkj6nfm/65B2DBSyhcTksD4H9nuGds/a3be81862610240ee35085e63835f531/007-web-1000px.jpg?fl=progressive&fm=jpg&w=1200"
        width={1000}
      />
    </picture>
  </span>
</div>
`;
